client/app/scripts/liveblog-users/module.js

```diff
@@ -0,0 +1,151 @@
+(function() {
+    'use strict';
+    ChangeAvatarController.$inject = ['$scope', 'upload', 'session', 'urls'];
+    function ChangeAvatarController($scope, upload, session, urls) {
+
+        $scope.methods = [
+            {id: 'upload', label: gettext('Upload from computer')}
+        ];
+
+        $scope.activate = function(method) {
+            $scope.active = method;
+            $scope.preview = {};
+            $scope.notify = {message:'', code: ''};
+            $scope.progress = {width: 0};
+        };
+
+        $scope.activate($scope.methods[0]);
+
+        $scope.upload = function(config) {
+            var form = {};
+            form.CropLeft = Math.round(Math.min(config.cords.x, config.cords.x2));
+            form.CropRight = Math.round(Math.max(config.cords.x, config.cords.x2));
+            form.CropTop = Math.round(Math.min(config.cords.y, config.cords.y2));
+            form.CropBottom = Math.round(Math.max(config.cords.y, config.cords.y2));
+
+            if (config.img) {
+                form.media = config.img;
+            } else if (config.url) {
+                form.URL = config.url;
+            } else {
+                return;
+            }
+
+            return urls.resource('upload').then(function(uploadUrl) {
+                return upload.start({
+                    url: uploadUrl,
+                    method: 'POST',
+                    data: form
+                }).then(function(response) {
+
+                    if (response.data._status === 'ERR'){
+                        return;
+                    }
+
+                    var picture_url = response.data.renditions.viewImage.href;
+                    $scope.locals.data.picture_url = picture_url;
+                    $scope.locals.data.avatar = response.data._id;
+
+                    return $scope.resolve(picture_url);
+                },  function(error) {
+                    $scope.notify.message = (error.statusText !== '') ? error.statusText : gettext('There was a problem with your upload');
+                    $scope.notify.code = 'error';
+                }, function(update) {
+                    $scope.progress.width = Math.round(update.loaded / update.total * 100.0);
+                });
+            });
+        };
+    }
+
+    angular.module('superdesk.users')
```

Comment:
Did you copy this module from superdesk ?
There is no other way to limit code duplication ? This is something that we want to avoid, when we update superdesk for instance :)